autoload/lsp/ui/vim/documentation.vim

```diff
@@ -6,7 +6,7 @@ let s:last_popup_id = -1
 function! s:complete_done() abort
     if !g:lsp_documentation_float | return | endif
     " Use a timer to avoid textlock (see :h textlock).
-    let l:event = deepcopy(v:event)
+    let l:event = copy(v:event)
     call timer_start(0, {-> s:show_documentation(l:event)})
```

Comment:
It should be debounced.

```viml
let s:last_timer_id = 0

function! s:complete_done() abort
    if !g:lsp_documentation_float | return | endif
    " Use a timer to avoid textlock (see :h textlock).
    let l:event = copy(v:event)
    call timer_stop(s:last_timer_id)
    let s:last_timer_id = timer_start(80, {-> s:show_documentation(l:event)})
endfunction
```

Reply:
Ok... but why?

Reply:
If we keep pressing `<C-n>` to choose item at the bottom of pum, the current implementation will send requests for each all items. It bit bad for performance.

Sorry. I know you don't touch it on this PR but it's good chance to improve it, I thought.

Reply:
Implemented. Please check and test.